clio-server/src/main/scala/org/broadinstitute/clio/server/service/IndexService.scala

```diff
@@ -67,4 +78,133 @@ abstract class IndexService[CI <: ClioIndex](
       elasticsearchIndex
     )
   }
+  private val keysToDrop =
+    Set(
+      ElasticsearchIndex.UpsertIdElasticsearchName,
+      ElasticsearchIndex.EntityIdElasticsearchName,
+    )
+
+  def validateUpsert(
```

Comment:
I think this is complicated enough that it needs a comment that describes what the code is doing and why.